### PR TITLE
Add ServerErrorHandler to inject dependencies in annotations

### PR DESCRIPTION
**Pull request by kth496**
Motivation:
- It seems good to directly inject ServerErrorHandler via a bean.
- https://github.com/line/armeria/issues/5440

Modifications:
- Introduced `Optional<List<ServerErrorHandler>>` serverErrorHandlers in the `armeriaServer` method to enable the injection of ServerErrorHandler beans.
- Updated the `configureServerWithArmeriaSettings` method to incorporate error handlers.

Result:
- Users can now define custom ServerErrorHandler beans and utilize an Armeria server that has been pre-configured with these user-defined error handlers.
- Closes https://github.com/line/armeria/issues/5443

### CODE REVIEW

File: examples/websocket/src/main/java/example/armeria/server/websocket/Main.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Main {
```
Comment (ikhoon) [inline examples/websocket/src/main/java/example/armeria/server/websocket/Main.java:1]: Remove?

File: spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java [hunk 2]
```diff
@@ -110,6 +111,7 @@ public static void configureServerWithArmeriaSettings(
             MeterIdPrefixFunction meterIdPrefixFunction,
             List<MetricCollectingServiceConfigurator> metricCollectingServiceConfigurators,
             List<DependencyInjector> dependencyInjectors,
+            List<ServerErrorHandler> serverErrorHandlers,
```
Comment (ikhoon) [inline spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaConfigurationUtil.java:114]: Should we also fix as well?https://github.com/line/armeria/blob/d89aee9b828a1feb6c5544ee2bf5c2c5aecd9fb8/spring/boot3-webflux-autoconfigure/src/main/java/com/linecorp/armeria/spring/web/reactive/ArmeriaReactiveWebServerFactory.java#L161-L161

**Comment by ikhoon** [comment]
Would you add test cases to check if injected error handlers are correctly performed?

**Comment by kth496** [comment]
@ikhoon Thanks for your feedback. I added some commits!

File: spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaAutoConfiguration.java [hunk 3]
```diff
@@ -152,7 +155,8 @@ public ArmeriaServerConfigurator gracefulShutdownServerConfigurator(
         if (GRACEFUL_SHUTDOWN.equalsIgnoreCase(shutdown)) {
             return sb -> sb.gracefulShutdownTimeout(duration, duration);
         } else {
-            return sb -> {};
+            return sb -> {
+            };
```
Comment (minwoox) [inline spring/boot3-autoconfigure/src/main/java/com/linecorp/armeria/spring/AbstractArmeriaAutoConfiguration.java:159]: nit: revert?

**Review by minwoox — APPROVED** [review]
Looks great! Thanks!

**Comment by ikhoon** [comment]
Could you check the lint failure? Everything else looks good.

**Comment by kth496** [comment]
@ikhoon Thanks, I fixed code styles and also checked the lint!

**Review by jrhee17 — APPROVED** [review]
Very nicely done 👍  Thanks @kth496 🙇 👍 🙇

**Review by ikhoon — APPROVED** [review]
Good to go! Thanks, @kth496. 🚀🚀

**Pull request merged**